cpp/include/Ice/IconvStringConverter.h

```diff
@@ -146,7 +146,7 @@ namespace IceInternal
         }
         catch (const Ice::IllegalConversionException& sce)
         {
-            throw Ice::IconvInitializationException(__FILE__, __LINE__, sce.reason());
+            throw Ice::IconvInitializationException(__FILE__, __LINE__, sce.what());
```

Comment:
Can we drop this exception and throw `Ice::InitializationException`?

Reply:
For a follow-up PR.